src/main/java/codeu/model/store/basic/MessageStore.java

```diff
@@ -70,6 +75,23 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
     return messages;
   }
 
+  /** Maps each message to its author. */
+  private void sortMessagesByAuthor(){
+    for (Message message : messages){
```

Comment:
Replace userMessages with an empty map so this doesn't end up containing old messages if setMessages() is called more than once.